haystack/nodes/other/join_docs.py

```diff
@@ -111,7 +123,10 @@ def _calculate_comb_sum(self, results):
 
         for result, weight in zip(results, weights):
             for doc in result:
-                scores_map[doc.id] += doc.score * weight
+                if self.sort_by_score:
+                    scores_map[doc.id] += doc.score * weight
+                else:
+                    scores_map[doc.id] += (doc.score if doc.score else 0) * weight
```

Comment:
I think we should keep only the second branch in this block.

Right now if `sort_by_score=True` but `doc.score=None`, the original issue arises again. However, if we change this block to:
```suggestion
                scores_map[doc.id] += (doc.score if doc.score else 0) * weight
```
The failure will never occur, but both `sort_by_score=True` and `sort_by_score=False` will still work as expected. Am I reading your code correctly?

Reply:
Sure, I can do that. I just wasn't sure whether you would like to keep the original behavior (where with `sort_by_score=False` you are getting an error thrown.)